jdbc-v2/src/main/java/com/clickhouse/jdbc/StatementImpl.java

```diff
@@ -212,8 +218,17 @@ public void setQueryTimeout(int seconds) throws SQLException {
 
     @Override
     public void cancel() throws SQLException {
-        checkClosed();
-        throw new UnsupportedOperationException("Cancel is not supported.");
+        if (closed) {
+            return;
+        }
+
+        try {
+            connection.client.query(String.format("KILL QUERY%sWHERE query_id = '%s'",
```

Comment:
https://clickhouse.com/docs/en/sql-reference/statements/kill#kill-query

there should be a cluster name. please see documentation